src/pages/images/ImageSelector.tsx

```diff
@@ -210,19 +210,25 @@ const ImageSelector: FC<Props> = ({ onSelect, onClose }) => {
           : item.variant;
 
       const getSource = () => {
-        if (item.created_at) {
-          return "Local";
+        let source = "Custom";
+        if (!item.cached && item.created_at) {
+          source = "Local";
         }
         if (item.server === canonicalServer) {
-          return "Ubuntu";
+          source = "Ubuntu";
         }
         if (item.server === minimalServer) {
-          return "Ubuntu Minimal";
+          source = "Ubuntu Minimal";
         }
         if (item.server === imagesLxdServer) {
-          return "LXD Images";
+          source = "LXD Images";
         }
-        return "Custom";
+        return (
+          <>
+            {source}{" "}
+            {item.cached ? <span className="u-text--muted">cached</span> : ""}
+          </>
+        );
       };
```

Comment:
Shall we keep this function to return the string as before. Then append the cached string based on the column below in the column? Seems a bit simpler to me.

Reply:
done :+1: